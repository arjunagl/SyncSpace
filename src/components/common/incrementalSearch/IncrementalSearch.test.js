import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import IncrementalSearchComponent, { IncrementalSearch } from './IncrementalSearch';


describe('<IncrementalSearch />', () => {
    afterEach(() => {
        //    this.constructorSpy.restore();
    });

    it('renders the component properly', () => {
        const component = shallow(<IncrementalSearch />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('calls componentDidMount', () => {
        const componentDidMountSpy = sinon.spy(IncrementalSearch.prototype, 'componentDidMount');
        mount(<IncrementalSearch />);
        expect(IncrementalSearch.prototype.componentDidMount.calledOnce).toEqual(true);
        componentDidMountSpy.restore();
    });

    it('calls \'onChange\' when the user types in something', () => {
        const onChangeSpy = sinon.spy(IncrementalSearch.prototype, 'onChange');
        const incrementalSearchWrapper = mount(<IncrementalSearch />);

        //find the input element
        const searchInput = incrementalSearchWrapper.find('#searchInput');
        searchInput.node.value = 'David';
        searchInput.simulate('change');
        expect(onChangeSpy.called).toEqual(true);
        onChangeSpy.restore();
    });

    it('calls \'onPerformIncrementalSearch\' when the user types in something', () => {
        const mockStore = configureStore();

        const onPerformIncrementalSearchSpy = sinon.spy();
        const mapStateToProps = null;
        const mapDispatchToProps = {
            onPerformIncrementalSearch: onPerformIncrementalSearchSpy
        };

        const mappedProps = { mapStateToProps, mapDispatchToProps };

        const incrementalSearchWrapper =
            shallow(
                <IncrementalSearchComponent
                    onPerformIncrementalSearch={onPerformIncrementalSearchSpy}
                    props={mappedProps}
                    store={mockStore}
                />
            );


        //find the input element
        const searchInput = incrementalSearchWrapper.find('#searchInput');
        searchInput.node.value = 'David';
        searchInput.simulate('change', searchInput);
        expect(onPerformIncrementalSearchSpy.called).toEqual(true);
        // onChangeSpy.restore();
    });
});
