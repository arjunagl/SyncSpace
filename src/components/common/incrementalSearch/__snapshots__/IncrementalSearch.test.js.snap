// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncrementalSearch /> renders the component properly 1`] = `
<div>
  <input
    id="searchInput"
    name="search"
    onChange={[Function]}
    placeholder="Search.."
    type="text"
  />
</div>
`;
